### PR TITLE
Change typed_routing.dart back to route_data.dart

### PR DESCRIPTION
**Pull request by johnpryan**
We can't change the name of this library since go_router_builder already has a published version that depends on it https://github.com/flutter/packages/pull/2381#discussion_r930151979

Fixes https://github.com/flutter/flutter/issues/108274

### CODE REVIEW

**Review by kevmoo — APPROVED** [review]

**Comment by stuartmorgan** [comment]
Please add the tooling roll to this PR (as described in the other PR's comments) so that we get presubmit validation that this is sufficient to fix everything.

**Review by chunhtai — APPROVED** [review]
LGTM
Do you plan to change it back at some point? If so, you should add a todo and probably does that as part of the go_router v5 breaking change

**Comment by chunhtai** [comment]
pinged Hixie on discord for NTE

**Review by loic-sharma — APPROVED** [review]

File: packages/go_router/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 4.2.2
+
+- Fixes a bug where go_router_builder wasn't detecting annotations
```
Comment (stuartmorgan) [inline packages/go_router/CHANGELOG.md:3]: Nit: Missing period (per style guide).

**Review by stuartmorgan — APPROVED** [review]
LGTM (with nit)

I've verified that this is running the tests with an override as expected:
https://cirrus-ci.com/task/6731406855372800?logs=pathified_unit_test#L3
https://cirrus-ci.com/task/6731406855372800?logs=pathified_unit_test#L256

**Comment by Hixie** [comment]
test-exempt: is tested: rolling flutter_plugin_tools will enable the test in ro_router_builder to use the tip of tree version of go_router which would have caught the error this is fixing.

**Pull request merged**